src/hotspot/share/runtime/vframeArray.cpp

```diff
@@ -446,23 +444,24 @@ void vframeArrayElement::unpack_on_stack(int caller_actual_parameters,
       thread->popframe_free_preserved_args();
     }
   }
-
 #ifndef PRODUCT
-  if (PrintDeoptimizationDetails) {
-    ttyLocker ttyl;
-    tty->print_cr("[%d. Interpreted Frame]", ++unpack_counter);
-    iframe()->print_on(tty);
-    RegisterMap map(thread,
-                    RegisterMap::UpdateMap::include,
-                    RegisterMap::ProcessFrames::include,
-                    RegisterMap::WalkContinuation::skip);
-    vframe* f = vframe::new_vframe(iframe(), &map, thread);
-    f->print();
-    if (WizardMode && Verbose) method()->print_codes();
-    tty->cr();
+  {
+    LogMessage(deoptimization) lm;
+    if (lm.is_debug()) {
+      NonInterleavingLogStream ls(LogLevelType::Debug, lm);
+      ls.print_cr("[%d. Interpreted Frame]", ++unpack_counter);
+      iframe()->print_on(&ls);
+      RegisterMap map(thread,
+                      RegisterMap::UpdateMap::include,
+                      RegisterMap::ProcessFrames::include,
+                      RegisterMap::WalkContinuation::skip);
+      vframe* f = vframe::new_vframe(iframe(), &map, thread);
+      f->print_on(&ls);
+      if (WizardMode && Verbose) method()->print_codes_on(&ls);
```

Comment:
I guess David already commented on this but we're trying to remove WizardMode and Verbose.  Can this be:
if (lm.is_trace()) method->print_codes_on(lm);
?
Also methodHandle doesn't require () because -> turns it into Method*.